### ISSUE TITLE
gcc not working on Windows

### ISSUE THREAD

**Issue by kjohnsen** [brian-team/brian2#1256]
I'm on Windows 10 and have been having compiler issues, so I tried switching to gcc, and that doesn't seem to work. 

In my `~/.brian/userpreferences`, when I have the line `codegen.cpp.compiler = 'gcc'` I get the following error:
```
KeyError                                  Traceback (most recent call last)
~\.conda\envs\brian2genn\lib\distutils\ccompiler.py in new_compiler(plat, compiler, verbose, dry_run, force)
   1007 
-> 1008         (module_name, class_name, long_description) = compiler_class[compiler]
   1009     except KeyError:
KeyError: 'gcc'
...further down the traceback...
~\.conda\envs\brian2genn\lib\distutils\ccompiler.py in new_compiler(plat, compiler, verbose, dry_run, force)
   1012             msg = msg + " with '%s' compiler" % compiler
-> 1013         raise DistutilsPlatformError(msg)
   1014 
DistutilsPlatformError: don't know how to compile C/C++ code on platform 'nt' with 'gcc' compiler
```

I definitely have `gcc` installed and in my path. Is there really a reason `gcc` wouldn't work on Windows?

**Comment by mstimberg** [comment]
Hi. @thesamovar might know more about this since I am not using Windows myself. But note that this message is not coming from Brian but from distutils which we use for the compiler setup. I think to use gcc on Windows you'd rather have to set the compiler to `mingw32` (see https://wiki.python.org/moin/WindowsCompilers#GCC_-_MinGW-w64_.28x86.2C_x64.29), but then we have hardcoded the name `gcc` in a number of places (e.g. when determining the compiler options), so I am not sure that would actually work. In general, I think there are a number of hurdles to get things working smoothly at least for the Cython code generation (see e.g. this [Stackoverflow answer](https://stackoverflow.com/a/57109148) ). The default workflow is to build C extensions with the same compiler as the Python interpreter itself, i.e. with msvc on Windows. It should be easier to support gcc for standalone, but I wouldn't say it's on our todo list currently. Pull requests welcome of course :smile:

**Comment by thesamovar** [comment]
We haven't supported gcc on windows for years because it caused a lot of headaches, e.g. for the reason Marcel mentioned. This might have changed with Python 3 but I don't know and definitely don't have bandwidth to look into it any time soon. ☹

I think I'm going to close this issue because I don't see us having time to look into this in the foreseeable future and we've not had any other reports of people not being able to use msvc compiler on windows. However, as Marcel says, if you have a pull request we can certainly review that!

**Comment by kjohnsen** [comment]
Sounds good. The only reason I ran into this in the first place was the MSVC C99 check issue I brought up and hopefully solved in #1257 anyway.
